scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# USAGE: standard-header-new_guards.pl filename.h >filename.h.new
+#
+# source-maintenance plungin to standardise the header guards
+#
+# Behaviour:
+# - files whose name does not end in .h are passed verbatim
+# - anything before the first #ifdef SOMETHING_ENDING_IN_H is copied verbatim
+# - the existing header guard is replaced with SQUID_RELATIVE_PATH_OF_FILE_H
+# - anything until the last #endif clause in the file is copied verbatim
+# - the last #endif clause is replaced with the closing header guard and an empty line
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+my $new_guard = $filename;
+$new_guard =~ s/[\/\-\.]/_/g;
+$new_guard = "SQUID_".uc($new_guard);
+
+unless ($filename =~ /\.h$/i) {
+    while (<>) { print; }
+    exit(0);
+}
+
+# print preamble
+while (<>) {
+    last if (/^#ifndef\s+(\w+_H)/);
+    print;
+}
+
+# we are at end of fine, no preamble found
+exit(0) unless defined($_);
+
+# need to rematch as $1 is in scope of loop
+/^#ifndef\s+(\w+_H)/;
+die("no header guard found") unless defined $1;
+my $current_guard = $1;
+<>;
+/^#define\s+(\w+)_H/;
+die("header new_guard mismatch") unless $1 eq $current_guard;
+
+print "#ifndef $new_guard\n#define $new_guard\n";
```

Comment:
For readability sake:

```suggestion
print "#ifndef $new_guard\n";
print "#define $new_guard\n";
```